tm2/pkg/sdk/auth/ante.go

```diff
@@ -139,11 +139,19 @@ func NewAnteHandler(ak AccountKeeper, bank BankKeeperI, sigGasConsumer Signature
 		stdSigs := tx.GetSignatures()
 
 		for i := 0; i < len(stdSigs); i++ {
+			var isNewAccount bool
 			// skip the fee payer, account is cached and fees were deducted already
 			if i != 0 {
 				signerAccs[i], res = GetSignerAcc(newCtx, ak, signerAddrs[i])
+				// only create new account when tx is a sponsor transaction
 				if !res.IsOK() {
-					return newCtx, res, true
+					if tx.IsSponsorTx() {
```

Comment:
i think it should be a combination of:
- is sponsored
- acocunt_number == 0
- pubkey not yet known

Reply:
great additions